src/app/zap-templates/templates/app/CHIPClientCallbacks-src.zapt

```diff
@@ -412,6 +415,48 @@ bool IMDefaultResponseCallback(const chip::app::Command * commandObj, EmberAfSta
     return true;
 }
 
+bool IMWriteResponseCallback(const chip::app::WriteClient * writeClient, EmberAfStatus status)
+{
+    ChipLogProgress(Zcl, "WriteResponse:");
+    LogStatus(status);
+
+    Callback::Cancelable * onSuccessCallback = nullptr;                                                                            \
+    Callback::Cancelable * onFailureCallback = nullptr;                                                                            \
+    NodeId sourceNodeId                      = writeClient->GetSourceNodeId();                                                     \
+    uint8_t seq                              = static_cast<uint8_t>(writeClient->GetAppIdentifier());                      \
```

Comment:
Why the change from intptr_t to uint64_t for the app identifier?  It appears that this is primarily used as a uint8 sequence number in this PR.  intptr_t seems more flexible, and should always be able to hold a uint8.

Reply:
Per discussions, when we use intptr_t, we are expecting the number to represent a pointer (object), during offline discussions, we will add the option to bind interaction model delegate to the Clients instead of the IinteractionModelEngline object.

In this case, uint64_t will be more suitable since the (stateful) object should be an instance of the delegate, and the number will be used by the apps that want to use a single delegate and will be a random number from the view of IM.

Here, the sequence number is one possible implementation.

Reply:
You are telling me that because the clients will be the delegate, they then have a pointer to `this` and, therefore, do not need another pointer.  But moving to uint64_t still constrains implementation style.

This seems like a tradeoff where flexibility is lost with no up side.